src/ctelnet.cpp

```diff
@@ -808,6 +808,8 @@ void cTelnet::sendTelnetOption(char type, char option)
 void cTelnet::replyFinished(QNetworkReply* reply)
 {
     mpProgressDialog->close();
+    packageDownloadReply->abort();
+    packageDownloadReply->deleteLater();
```

Comment:
I do not think it is safe to use `QNetworkReply::deleteLater()` until AFTER we have extracted the data from the reply in the line:
```cpp
    file.write(reply->readAll());
```
below...!

Reply:
Why not? https://doc.qt.io/qt-5/qobject.html#deleteLater `The object will be deleted when control returns to the event loop.`

Reply:
You don't want to use `close`/`deleteLater` until you have gotten the data out - I am never sure when the event loop is returned to.

It just makes logical sense to put either of those under the chunk of code (in the same method remember) that reads the data otherwise it might go away before hand.

🤔 Looking more closely {see: https://doc.qt.io/qt-5/qnetworkreply.html#abort } - to avoid reading a partial reply that had been aborted - it would be wise to gate the reading of that data with a check that the error state is `QNetworkReply::NoError` or at least NOT `QNetworkReply::OperationCanceledError` - because **calling `close()` or `abort()` on the `QNetwordReply` will *also* raise the `QNetworkReply::finished()` signal that is connected to this slot**!

Reply:
I'll change the order, but "I am never sure when the event loop is returned to." - well never in the middle of the function, only after it is done at best, so it's safe. Try it out - you'll see that it won't crash.

Reply:
What about the method being called after an abort/cancel...

Reply:
Even then, it won't go off in the middle of the function, that is not possible